packages/contracts-ops/contracts/test/RebootVerify.t.sol

```diff
@@ -0,0 +1,256 @@
+        domain = "milkomedaC1";
+        uint256 blockNumber = 7355537;
+        domainNumber = getDomainNumber(domain);
+        vm.createSelectFork(vm.envString("RPC_URL_MILKOMEDAC1"), blockNumber);
+
+        verifyConfigImplementations();
+        readGovActions();
+        executeGovActions();
+        verifyGovActions();
+    }
+
+    /// @notice Verify that the implementations have been deployed at the addresses that exist
+    /// in the configuration file.
+    /// @dev To verify, we either call some constant that exists in the implementation, or a function that we
+    /// expect to exist
+    function verifyConfigImplementations() internal {
+        // verify home
+        homeImpl = Home(homeUpgrade(domain).implementation);
+        assertEq(homeImpl.MAX_MESSAGE_BODY_BYTES(), 2 * 2**10);
+        // verify replica
+        string memory remote = getConnections(domain)[0];
+        replicaImpl = Replica(replicaOfUpgrade(domain, remote).implementation);
+        assertEq(replicaImpl.LEGACY_STATUS_PROVEN(), bytes32(uint256(1)));
+        // verify gov router
+        govRouterImpl = GovernanceRouter(
+            governanceRouterUpgrade(domain).implementation
+        );
+        govRouterImpl.localDomain();
+        // verify bridge router
+        bridgeRouterImpl = BridgeRouter(
+            payable(bridgeRouterUpgrade(domain).implementation)
+        );
+        assertEq(bridgeRouterImpl.DUST_AMOUNT(), 0.06 ether);
+        // verify bridge token
+        bridgeTokenImpl = BridgeToken(
+            bridgeTokenUpgrade(domain).implementation
+        );
+        assertEq(
+            bridgeTokenImpl._PERMIT_TYPEHASH(),
+            keccak256(
+                "Permit(address owner,address spender,uint256 value,uint256 nonce,uint256 deadline)"
+            )
+        );
+        //verify token registry
+        tokenRegistryImpl = TokenRegistry(
+            tokenRegistryUpgrade(domain).implementation
+        );
+        tokenRegistryImpl.tokenBeacon();
+    }
+
+    /// @notice Verify the on-chain effects of executing the gov actions in the fork
+    function verifyGovActions() internal {
+        // Verify Updater rotation
+        UpdaterManager updaterManager = getUpdaterManager(domain);
+        assertEq(updaterManager.updater(), getUpdater(domain));
+
+        // Verify Replica enrollment
+        XAppConnectionManager cnMgr = getXAppConnectionManager(domain);
+        string[] memory connections = getConnections(domain);
+        for (uint256 i; i < connections.length; i++) {
+            uint32 connectionDomain = getDomainNumber(connections[i]);
+            address replica = address(getReplicaOf(domain, connections[i]));
+            assertEq(cnMgr.replicaToDomain(replica), uint256(connectionDomain));
+            assertEq(cnMgr.domainToReplica(connectionDomain), replica);
+        }
+
+        // Verify new implementations have been set
+
+        // Verify home implementation
+        address storedHomeImpl = vm
```

Comment:
bit easier to read to refactor these into a helper function which takes in a beacon + an expected implementation address